std_assert_test.go

```diff
@@ -0,0 +1,249 @@
+		return a.fail(t, "expected no error, got: %v", err)
+	}
+	return true
+}
+func (a fatal) NoError(t *testing.T, err error, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	if err != nil {
+		return a.fail(t, "expected no error, got: %v", err)
+	}
+	return true
+}
+
+func (a nonfatal) Error(t *testing.T, err error, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	if err == nil {
+		return a.fail(t, "expected error, got nil")
+	}
+	return true
+}
+func (a fatal) Error(t *testing.T, err error, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	if err == nil {
+		return a.fail(t, "expected error, got nil")
+	}
+	return true
+}
+
+func (a nonfatal) Panics(t *testing.T, f func(), msgAndArgs ...interface{}) bool {
+	t.Helper()
+	defer func() {
+		if r := recover(); r == nil {
+			_ = a.fail(t, "expected panic, but function did not panic")
+		}
+	}()
+	f()
+	return true
+}
+func (a fatal) Panics(t *testing.T, f func(), msgAndArgs ...interface{}) bool {
+	t.Helper()
+	defer func() {
+		if r := recover(); r == nil {
+			_ = a.fail(t, "expected panic, but function did not panic")
+		}
+	}()
+	f()
+	return true
+}
+
+func (a nonfatal) Empty(t *testing.T, object interface{}, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	if !isEmpty(object) {
+		return a.fail(t, "expected empty, got: %#v", object)
+	}
+	return true
+}
+func (a fatal) Empty(t *testing.T, object interface{}, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	if !isEmpty(object) {
+		return a.fail(t, "expected empty, got: %#v", object)
+	}
+	return true
+}
+
+func isEmpty(object interface{}) bool {
+	if object == nil {
+		return true
+	}
+	v := reflect.ValueOf(object)
+	switch v.Kind() {
+	case reflect.Array, reflect.Slice, reflect.Map, reflect.Chan, reflect.String:
+		return v.Len() == 0
+	case reflect.Ptr, reflect.Interface:
+		if v.IsNil() {
+			return true
+		}
+		return isEmpty(v.Elem().Interface())
+	}
+	// numbers and structs are never considered empty here
+	return false
+}
+
+func (a nonfatal) Len(t *testing.T, object interface{}, length int, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	v := reflect.ValueOf(object)
+	switch v.Kind() {
+	case reflect.Array, reflect.Slice, reflect.Map, reflect.Chan, reflect.String:
+		if v.Len() != length {
+			return a.fail(t, "unexpected length, expected %d got %d", length, v.Len())
+		}
+		return true
+	default:
+		return a.fail(t, "Len not supported for kind %s", v.Kind())
+	}
+}
+func (a fatal) Len(t *testing.T, object interface{}, length int, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	v := reflect.ValueOf(object)
+	switch v.Kind() {
+	case reflect.Array, reflect.Slice, reflect.Map, reflect.Chan, reflect.String:
+		if v.Len() != length {
+			return a.fail(t, "unexpected length, expected %d got %d", length, v.Len())
+		}
+		return true
+	default:
+		return a.fail(t, "Len not supported for kind %s", v.Kind())
+	}
+}
+
+// Allow formatting helpers similar to fmt in tests using fmt.Sprintf
+var _ = fmt.Sprintf
+var _ = runtime.NumCPU
```

Comment:
I don't see a need for this.

I feel like you are forcing import. So symbols are available. 

Why that ?

I'm not saying you are wrong. I'm saying I'm suspicious